simulator-spring-boot/src/main/java/org/citrusframework/simulator/http/HttpOperationScenarioRegistrar.java

```diff
@@ -19,10 +37,15 @@ public class HttpOperationScenarioRegistrar implements OpenApiSpecificationProce
     public void process(OpenApiSpecification openApiSpecification) {
 
         HttpScenarioGenerator generator = new HttpScenarioGenerator(openApiSpecification);
-        ApplicationContext applicationContext = SimulatorConfigurationProperties.getApplicationContext();
 
-        if (applicationContext instanceof ConfigurableApplicationContext configurableApplicationContext) {
-            generator.postProcessBeanFactory(configurableApplicationContext.getBeanFactory());
-        }
+        CitrusInstanceManager.get().ifPresent(citrus -> {
+            ReferenceResolver referenceResolver = citrus.getCitrusContext().getReferenceResolver();
+            if (referenceResolver instanceof SpringBeanReferenceResolver springBeanReferenceResolver) {
```

Comment:
this could be:

```suggestion
            if (referenceResolver instanceof SpringBeanReferenceResolver springBeanReferenceResolver
                && springBeanReferenceResolver.getApplicationContext() instanceof AbstractApplicationContext abstractApplicationContext) {
```

Reply:
😮